addons/transifex/models/ir_translation.py

```diff
@@ -8,17 +8,64 @@
     from ConfigParser import ConfigParser
 from os.path import join as opj
 import os
+import re
 import werkzeug
 
-from odoo import models, fields
+from odoo import api, models, fields, _
 from odoo.modules.module import ad_paths
 
+# detect placeholders %s, %d, %(foo)s,...
+REGEX_PLACEHOLDERS = re.compile("%(\(\w+\)|)([0+-]|)(\d+|\*|)[diouxXcrsa]|%(\(\w+\)|)([0+-]|)(\d+|\*|)(\.(\d+|\*|)|)[eEfFgG]|%%")
+REGEX_XML_TAGS = re.compile("</?(\w+)")
 
 class IrTranslation(models.Model):
 
     _inherit = 'ir.translation'
 
     transifex_url = fields.Char("Transifex URL", compute='_get_transifex_url')
+    is_valid = fields.Boolean(compute='_compute_validity')
+
+    @api.depends('src', 'value')
+    def _compute_validity(self):
+        bad = self._get_validity()
+        bad.update({'is_valid': False})
+        good = self - bad
+        good.update({'is_valid': True})
+
+    def _get_validity(self):
+        bad = self.browse()
+        for translation in self:
+            if not translation.value:
+                continue
+
+            # detect placeholders mismatches
+            placeholders_src = sorted([m.group(0) for m in REGEX_PLACEHOLDERS.finditer(translation.src)])
+            placeholders_trans = sorted([m.group(0) for m in REGEX_PLACEHOLDERS.finditer(translation.value)])
+            if placeholders_src != placeholders_trans:
```

Comment:
Alternatively, sort by `m[1]` or `(m[1] and m[0])`, this way the order doesn't change if the placeholders don't have a key (as they should all yield None), since positional placeholders can't be reordered.